header-rewrite-filter/header_processor.cc

```diff
@@ -547,6 +588,11 @@ namespace HeaderRewriteFilter {
             source = std::get<1>(result);
             return std::make_tuple(absl::OkStatus(), source);
         }
+        case Utility::FunctionType::Static:
+        {
+            source = function_argument_;
```

Comment:
Nit: `source` is never used later, so you could remove the declaration above, and always declare these const right where they're used, e.g.
`const std::string source = function_argument_;`

+1 on declaring the variable here though, giving it the name `source` is great self-documenting.

Reply:
If you want to be really efficient, you can use [`std::move`:](https://en.cppreference.com/w/cpp/utility/move)

```
const std::string source = function_argument_;
std::make_tuple(absl::OkStatus(), std::move(source));
```

You have to be a bit careful with `move`, after calling it `source` is no longer valid and can never be used. But you don't here, we very quickly return and end the scope. And it lets you avoid the copy.

(_How_ it works is very complicated, I admittedly barely understand it. Technically, it produces an _xvalue_. But you can think of `std::move` as magic to avoid the copy, as long as you never use the moved variable again. If you really want to understand it, and don't value your sanity, there's an explanation of `move` [here](https://stackoverflow.com/questions/7510182/how-does-stdmove-transfer-values-into-rvalues) and of rvalue/lvalue/xvalue [here](https://en.cppreference.com/w/cpp/language/value_category).

Reply:
Ahh yes -- there's still places in the code where I am reusing variables. Will fix this and make sure all the other places are also changed in a code cleanup PR.

Reply:
Cleaned this up in #36